### PR TITLE
#5526: fix duplicate entries in workshop type filter

### LINKED ISSUE #5526: Brick listed 2x in workshop filter type (one entry is for readers)
image

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Closes #5526 
- Fixes the type of `EditablePackage.kind` (see contract.ts). The values are the display values here: 
  - https://github.com/pixiebrix/pixiebrix-app/blob/be1c486eba393e3c8e2f99401f78af5958b4060b/api/models/registry.py#L210-L210
  - https://github.com/pixiebrix/pixiebrix-app/blob/be1c486eba393e3c8e2f99401f78af5958b4060b/api/serializers/brick.py#L65-L65 
- Fixes the workshop filter by mapping the values to not distinguish bricks vs. readers

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer: @mnholtz

### CODE REVIEW

File: src/extensionConsole/pages/workshop/WorkshopPage.tsx [hunk 1]
```diff
@@ -89,9 +91,11 @@ function useSearchOptions(bricks: EnrichedBrick[]) {
 
   const kindOptions = useMemo(
     () =>
-      sortBy(compact(uniq((bricks ?? []).map((x) => x.kind)))).map((value) => ({
+      sortBy(
+        compact(uniq((bricks ?? []).map((x) => mapKindToKindUiValue(x.kind))))
```
Comment (grahamlangford) [inline src/extensionConsole/pages/workshop/WorkshopPage.tsx:95]: I don't think `compact` serves any purpose here anymore. `mapKindToKindUiValue` always returns a `KindFilterValue`, so there are no falsey values to remove.

**Review by grahamlangford — APPROVED** [review]

**Comment by twschiller** [comment]
Keeping open for @mnholtz to review since this is correcting the backend contract

**Review by johnnymetz — APPROVED** [review]

File: src/types/contract.ts
Comment (mnholtz) [inline src/types/contract.ts:1]: @twschiller I'd like to see a description added to kind further up in this file: https://github.com/pixiebrix/pixiebrix-extension/blob/ee3d439d5ad6ef23426042a84a4b9009e1e91ae3/src/types/contract.ts#L43-L49

It's not super clear what the `Kind` type is used for in this case.

File: src/extensionConsole/pages/workshop/workshopUtils.ts [hunk 2]
```diff
@@ -15,27 +15,44 @@
...
-  ["block", "Brick"],
-  ["reader", "Brick"],
-  ["blueprint", "Mod"],
-  ["recipe", "Mod"],
-  ["service", "Integration"],
-  // Full name is "starter brick", but user shorter name
-  ["foundation", "Starter"],
-]);
+/**
+ * A valid kind value.
```
Comment (mnholtz) [inline src/extensionConsole/pages/workshop/workshopUtils.ts:21]: nit: Valid in what context?

File: src/extensionConsole/pages/workshop/WorkshopPage.tsx [hunk 3]
```diff
@@ -113,7 +117,7 @@ const CustomBricksSection: React.FunctionComponent<NavigateProps> = ({
     isLoading,
     error,
   } = useGetEditablePackagesQuery(undefined, {
-    // Make sure user always see the latest bricks available
+    // Make sure user always see the latest bricks available (e.g., because they just saved a mod in the page editor)
```
Comment (mnholtz) [inline src/extensionConsole/pages/workshop/WorkshopPage.tsx:120]: ❤️

**Review by mnholtz — APPROVED** [review]
Approved with some requests for additional/improved comments to describe types.

File: src/types/contract.ts
Comment (twschiller) [reply src/types/contract.ts:1]: @mnholtz I've updated and wrote on RegistryPackage. This type was the lowercase union of what would come back for package details and editable package metadata :(

Comment (mnholtz) [reply src/types/contract.ts:1]: @twschiller great, I took a look at the changes. That's a lot better.

**Pull request merged**